src/libtorch.cc

```diff
@@ -786,7 +811,8 @@ ModelInstanceState::Execute(
     torch::jit::setGraphExecutorOptimize(
         !model_state_->DisabledOptimizedExecution());
 
-    torch::NoGradGuard no_grad;
+    // enable/disable inference mode
+    torch::InferenceMode infer_guard(model_state_->InferenceMode());
```

Comment:
Shouldn't we fallback to NoGradGuard if INFERENCE_MODE is set false. I mean that is the current default behavior. 
According to the numbers you shared it looked like in some cases NoGradGuard was giving the best performance.

Reply:
You are right. Fixing this now.

Reply:
So if infer model is set, the no_grad below will be ignored?

Reply:
Yes. When enabled, InferenceMode enables NoGrad. Technically NoGrad is a subset of InferenceMode